mixins/withStyles.mdx

```diff
@@ -0,0 +1,146 @@
+> `class Box extends withStyles(class extends lng.Component {...}, styles) {}`
+> This may seem a little gross, but we are just doing a shorthand here. You could always define a base class outside of `withStyles` and pass it in. Another option would be `class Box extends withStyles(lng.Component, styles) {}`, which seems a lot cleaner, BUT -- this will pollute the `lng.Component` prototype _and_ static class! If you want _every_ Lightning component in your app to inherit `styles` and `theme`, go for it (but we do not recommend this)!
+
+Composing a `BlueBox` becomes much simpler now:
+
+```js
+const blueStyles = { color: 0xff0000ff }
+class BlueBox extends withStyles(Box, blueStyles) {}
+```
+
+My `BlueBox` will maintain the `w` and `h` properties of `Box` because `withStyles` merges the new `styles` object with that of the base class.
+
+> The `styles` object can be any shape you want. There are no hard rules there, but it makes sense to use property names that are representative of their values. The implementation of `styles` values in your class declaration is also up to you. Remember: if you use `withStyles`, you can use it for as little or as many properties as you'd like, although it's recommended to use it for _all_ properties that are part of styling.
+
+With this refactor, my `Box` properties can still be used normally
+
+```js
+// works
+class Example extends lng.Component {
+  static _template() {
+    return {
+      Box: {
+        type: Box,
+        w: 150,
+        h: 50,
+        color: 0xff00ff00
+      }
+    }
+  }
+}
+```
+
+### Usage with Theme
+
+<Canvas>
+  <Story id="mixins-withstyles--with-theme" />
+</Canvas>
+
+So far, we have seen `withStyles` implementations using a `styles` object as an argument. Another way to implement `withStyles` is to pass a `styles` function that accepts a `theme` object as an argument. See the [Themes](?path=/docs/styles-themes--page) documentation for more info on theme structure.
+
+```js
+const theme = {
+  colors: {
+    background: 0xff000000,
+    text: 0xffffffff
+  }
+};
+
+const styles = theme => {
+  color: theme.colors.background,
+  textColor: theme.colors.text
+};
+
+class Box extends withStyles(class extends lng.Component {
+  static _template() {
+    return {
+      w: 100,
+      h: 100,
+      color: this.styles.color,
+      rect: true
+    }
+  }
+}, styles, theme) { }
+
+class Text extends withStyles(class extends lng.Component {
+  static _template() {
+    return {
+      text: {
+        textColor: this.styles.textColor
+      }
+    }
+  }
+}, styles, theme) { }
+```
+
+In this arbitrary example, we can see that both `Box` and `Text` are pulling from the same `styles` and using the same `theme`. We can imagine exporting a `theme` object that allows setting stylistic values at a global level, while `styles` objects could be set separately at a component level. In fact, that is exactly what we are doing in @lightning/ui!
```

Comment:
Should remove this last sentence because it references `@lightning/ui`. I'll look for other references that should be removed too